Sources/Automerge/ActorId.swift

```diff
@@ -1,15 +1,20 @@
 import AutomergeUniffi
+import Foundation
 
 /// The identifier for collaborators contributing to an Automerge document.
 ///
 /// Each separate instance of an Automerge document should have it's own, unique, `ActorId`.
 /// If you create your own `ActorId`, no concurrent changes should ever be made with the same `ActorId`.
 public struct ActorId: Equatable, Hashable, Sendable {
-    var bytes: [UInt8]
+    var data: Data
+
+    public init(data: Data) {
```

Comment:
If we're going to have an initializer, I'd prefer to have a default initializer that doesn't require any input, to minimize the use of it - but allows you to pass in a 16-byte data buffer to support your use case and needs. I think it also needs to be a failable initializer that throws an error if it's other than 16 bytes.

Since explicit length arrays are kind of a PITA in swift, I'd recommend something like:


```swift
public init?(data: Data? = nil) {
// ... check for explicitly 16 bytes, and throw an error if its not
}
```

I'm asserting this based on the idea that the Automerge core _doesn 't_ support anything other than 16 byte actorIds, but I'm not 100% sure on that point. @alexjg would be best to weigh in there. I think that's what I read on the Rust source, but I'm not super confident that I didn't miss some alternate trait that allows for variable lengths to be used somewhere.

Reply:
We're currently using 32-byte actorIds (since they are a concatenation of 2 UUIDs), so I would hope the newer versions of the Rust code isn't enforcing 16 bytes for backwards compatibility at least. I did a quick look through the Rust code and do see that the actor ID is specified as 16 bytes, but also see a case where it's being set to 20, so doesn't seem like a hard limit at least.

Reply:
Let's definitely start by adding a test to make sure it round trips from initializer to returning a full 32 bytes for your example setup, and we can pin down a "reasonable max size" after. If not, it'll likely mean core work in the Rust library to accommodate the size differences before we can finalize what you need here in the Swift overlay.

If you don't mind, check the PR to allow for updates, and I'll help add documentation and such to this PR, rather than trying to slap it in after the fact - as well as some of the relevant tests that I'd like to see, at least if you're amenable.

Reply:
> Let's definitely start by adding a test to make sure it round trips from initializer to returning a full 32 bytes for your example setup, and we can pin down a "reasonable max size" after. If not, it'll likely mean core work in the Rust library to accommodate the size differences before we can finalize what you need here in the Swift overlay.

Updated with a test.

> If you don't mind, check the PR to allow for updates, and I'll help add documentation and such to this PR, rather than trying to slap it in after the fact - as well as some of the relevant tests that I'd like to see, at least if you're amenable.

Is this something I need to set on the PR? Or you just mean to merge suggested changes?

Reply:
This detail: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork

I'm returning to Seattle tomorrow, so I won't add anything straight-away, but I thought I might add docs and tests as commits to your pull request branch, if you allow it.

Reply:
The `ActorId` struct is not hardcoded to a specific lenght, however if the actor ID is 16 bytes or less then it can be stack allocated in many cases which has a small performance impact.

Reply:
no worries @bgomberg - I'll work on putting suggestions into the PR review process itself. It's hard to ask for added lines, etc - but we'll work it out, and I can do additional docs to what I'd like to see after the fact if needed.